src/sentry/api/endpoints/client_state.py

```diff
@@ -0,0 +1,52 @@
+# This endpoint helps managing persisted client state with a TTL for a member, organization or user
+
+from django.conf import settings
+from rest_framework.exceptions import NotFound
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.utils import json, redis
+
+STATE_CATEGORIES = {
+    "onboarding": {
+        "ttl": 30 * 24 * 60 * 60,  # the time in seconds that the state will be persisted
+        "scope": "member",  # Can be "org" or "member"
+        "max_payload_size": 1024,
+    }
+}
+
+
+class ClientStateEndpoint(OrganizationEndpoint):
+    private = True
+
+    def __init__(self, **options) -> None:
+        cluster_key = getattr(settings, "SENTRY_CLIENT_STATE_REDIS_CLUSTER", "default")
+        self.client = redis.redis_clusters.get(cluster_key)
+        super().__init__(**options)
+
+    def get_key(self, organization, category, user):
+        if category not in STATE_CATEGORIES:
+            raise NotFound(detail="Category not found")
+        scope = STATE_CATEGORIES[category]["scope"]
+        if scope == "member":
+            return f"client-state:{category}:{organization}:{user}"
+        elif scope == "org":
+            return f"client-state:{category}:{organization}"
+
+    def get(self, request: Request, organization, category) -> Response:
+        key = self.get_key(organization.slug, category, request.user)
+        res = self.client.get(key)
+        if res:
+            self.client.expire(key, STATE_CATEGORIES[category]["ttl"])
```

Comment:
@Neo-Zhixing should we expire the cache when we load it? IMO the cache is valid until it's re-written.

Reply:
EXPIRE in the context of redis really means `set TTL`
https://www.redis.io/commands/expire/

Reply:
@Neo-Zhixing I see, just seems weird that reading would have this side effect. Aren't we setting the TTL already when we read it? I guess the purpose of this is to extend the TTL then, yea? If so, I think we should remove it. Otherwise, we could have cache values that never expire if they keep getting read but I don't think we really want that.

Reply:
good point. removed.